### PR TITLE
Fix 5577

### PR DESCRIPTION
**Pull request by jonesh66**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: examples/wdio/mocha/mocha.test.js [hunk 1]
```diff
@@ -1,6 +1,4 @@
 describe('webdriver.io page', () => {
-    it('should be a pending test')
```
Comment (christian-bromann) [inline examples/wdio/mocha/mocha.test.js:2]: can we revert this change? It has nothing to do with the actual issue.

File: examples/wdio/mocha/wdio.conf.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ exports.config = {
 
     mochaOpts: {
         ui: 'bdd',
-        timeout: 5000
+        timeout: 5000,
```
Comment (christian-bromann) [inline examples/wdio/mocha/wdio.conf.js:36]:
```suggestion
        timeout: 5000
```

File: packages/wdio-reporter/src/index.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import HookStats, { Hook } from './stats/hook'
 import TestStats, { Test } from './stats/test'
 import RunnerStats, { Runner } from './stats/runner'
 import { AfterCommandArgs, BeforeCommandArgs } from './types'
+import * as fs from 'fs'
```
Comment (christian-bromann) [inline packages/wdio-reporter/src/index.ts:10]:
```suggestion
import fs from 'fs'
```

Comment (christian-bromann) [reply packages/wdio-reporter/src/index.ts:10]: Also can we move up the import to keep native modules first?`

File: packages/wdio-reporter/src/index.ts [hunk 4]
```diff
@@ -40,6 +41,7 @@ export default class WDIOReporter extends EventEmitter {
     }
     retries = 0
     runnerStat?: RunnerStats
+    contentPresent: boolean
```
Comment (christian-bromann) [inline packages/wdio-reporter/src/index.ts:44]: How about renaming this flag to `isContentPresent` ?

File: packages/wdio-reporter/tests/reporter.test.ts [hunk 5]
```diff
@@ -65,6 +65,13 @@ describe('WDIOReporter', () => {
         expect(options.writeStream.write).toBeCalledWith('foobar')
     })
 
+    it('should set contentPresent to true when content is passed to write()', () => {
+        const options = { stdout: true, writeStream: { write: jest.fn() } as unknown as WriteStream }
+        const reporter = new WDIOReporter(options)
+        reporter.write('foobar')
```
Comment (christian-bromann) [inline packages/wdio-reporter/tests/reporter.test.ts:71]: Let's double check that the flag is false before
```suggestion
        expect(reporter.contentPresent).toBe(false)
        reporter.write('foobar')
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Couple of suggestions

File: examples/wdio/mocha/mocha.test.js (hunk 1 above)
Comment (jonesh66) [reply examples/wdio/mocha/mocha.test.js]: Done.

File: packages/wdio-reporter/src/index.ts (hunk 4 above)
Comment (jonesh66) [reply packages/wdio-reporter/src/index.ts]: Done in the latest commit.

**Comment by jonesh66** [comment]
@christian-bromann all requested changes are in place(as far as I can see).

File: packages/wdio-reporter/tests/reporter.test.ts [hunk 6]
```diff
@@ -65,6 +65,14 @@ describe('WDIOReporter', () => {
         expect(options.writeStream.write).toBeCalledWith('foobar')
     })
 
+    it('should set contentPresent to true when content is passed to write()', () => {
+        const options = { stdout: true, writeStream: { write: jest.fn() } as unknown as WriteStream }
+        const reporter = new WDIOReporter(options)
+        expect(reporter.contentPresent).toBe(false)
```
Comment (christian-bromann) [inline packages/wdio-reporter/tests/reporter.test.ts:71]:
```suggestion
        expect(reporter.isContentPresent).toBe(false)
```

**Review by christian-bromann — COMMENTED** [review]
almost all 😉

**Comment by christian-bromann** [comment]
@jonesh66 would you mind to update your branch. It would fix the failing pipeline.

**Comment by jonesh66** [comment]
@christian-bromann I merged my branch with master and smoke tests have failed on my branch. Looking into it.

**Comment by jonesh66** [comment]
Screenshot 2020-11-15 at 7 10 53 PM

@christian-bromann there's one failing test that's both on master and on this branch which might be unrelated to these changes

**Comment by christian-bromann** [comment]
> there's one failing test that's both on master and on this branch which might be unrelated to these changes

The current master branch seems to pass at the moment. This looks like a typescript issue.

**Comment by christian-bromann** [comment]
@jonesh66 is there anything I can do to help getting this merged?

**Comment by jonesh66** [comment]
@christian-bromann I haven't been able to point out where the failures occur based on the logs in the smoke suite. I am merging with the latest master. It would be helpful if you could point me to the failed tests

File: packages/wdio-reporter/src/index.ts [hunk 7]
```diff
@@ -40,6 +41,7 @@ export default class WDIOReporter extends EventEmitter {
     }
     retries = 0
     runnerStat?: RunnerStats
+    isContentPresent: boolean
```
Comment (christian-bromann) [inline packages/wdio-reporter/src/index.ts:44]:
```suggestion
    isContentPresent = false
```

File: packages/wdio-reporter/src/index.ts [hunk 8]
```diff
@@ -54,6 +56,7 @@ export default class WDIOReporter extends EventEmitter {
             : createWriteStream((this.options as WDIOReporterOptionsFromLogFile).logFile)
 
         let currentTest: TestStats
+        this.isContentPresent = false
```
Comment (christian-bromann) [inline packages/wdio-reporter/src/index.ts:59]:
```suggestion
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
We can set the default value directly in the class head like this.

**Comment by jonesh66** [comment]
@christian-bromann all changes are in place. Thank you for the support.

**Review by christian-bromann — APPROVED** [review]
Thanks for the contribution 👍
